x-pack/agent/pkg/agent/application/fleet_acker.go

```diff
@@ -0,0 +1,92 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package application
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/elastic/beats/x-pack/agent/pkg/agent/errors"
+	"github.com/elastic/beats/x-pack/agent/pkg/core/logger"
+	"github.com/elastic/beats/x-pack/agent/pkg/fleetapi"
+	"github.com/elastic/beats/x-pack/agent/pkg/scheduler"
+)
+
+const (
+	// EventTypeAction is an event type describing action related events.
+	EventTypeAction = "ACTION"
+	// EventSubtypeACK is an ACTION event subtype specifying ACK event.
+	EventSubtypeACK = "ACKNOWLEDGED"
+)
+
+type actionAcker struct {
+	log        *logger.Logger
+	dispatcher dispatcher
+	client     clienter
+	scheduler  scheduler.Scheduler
+	agentInfo  agentInfo
+	reporter   fleetReporter
+	done       chan struct{}
+}
+
+func newActionAcker(
+	log *logger.Logger,
+	agentInfo agentInfo,
+	client clienter,
+) (*actionAcker, error) {
+	return &actionAcker{
+		log:       log,
+		client:    client,
+		agentInfo: agentInfo, //TODO(ph): this need to be a struct.
+	}, nil
+}
+
+func (f *actionAcker) Ack(actionID string) error {
+	// get events
+	event := newActionEvent(actionID)
+
+	// checkin
+	cmd := fleetapi.NewCheckinCmd(f.agentInfo, f.client)
+	req := &fleetapi.CheckinRequest{
+		Events: []fleetapi.SerializableEvent{event},
+	}
+
+	_, err := cmd.Execute(req)
+	if err != nil {
+		return errors.New(err, fmt.Sprintf("acknowledge action '%s' failed", actionID), errors.TypeNetwork)
+	}
+
+	return nil
+}
```

Comment:
Since we received Action a batch I think we should `Ack` them in batch if possible to reduce the round trip and resources from the Kibana instance.

Reply:
i think this is in conflict with this PR,
if handler is responsible to ACK then it will ACK one action at a time. 
if we want to batch ACKs we need to distinguish between ACKs which needs to be fired immediately (e.g restart ack) and actions which can be batched. then we would need to provide Acker with some queue of ACKs and logic to differentiate between immediate ACKs and delayed.

Reply:
Ok, let's keep the ack immediate for now, but my comment concerning sending the ACKer in the dispatch instead of having a field in the dispatcher is still valid?

Reply:
yep we can do that